Tests/Unit/Controller/TeaControllerTest.php

```diff
@@ -51,6 +58,11 @@ protected function setUp(): void
         /** @var TeaRepository&ProphecySubjectInterface $teaRepository */
         $teaRepository = $this->teaRepositoryProphecy->reveal();
         $this->subject->injectTeaRepository($teaRepository);
+
+        $this->responseFactoryProphecy = $this->prophesize(ResponseFactory::class);
+        /** @var ResponseFactory&ProphecySubjectInterface $responseFactory */
+        $responseFactory = $this->responseFactoryProphecy->reveal();
```

Comment:
`$this->subject->injectTeaRepositoy()` is changed to return `$this->htmlResponse()` .. which is wrong anyway. You added code to inject the responseFactory .. but "to late". The responseFactory must be existing in the actionController before first usage (which is added in `injectTeaRepository` ...